govcd/system.go

```diff
@@ -921,3 +921,24 @@ func (vcdClient *VCDClient) GetAdminOrgByNameOrId(identifier string) (*AdminOrg,
 	}
 	return entity.(*AdminOrg), err
 }
+
+// Returns the UUID part of an HREF
+// Similar to getBareEntityUuid, but tailored to HREF
+func GetUuidFromHref(href string) (string, error) {
```

Comment:
A few questions - do we really need this to be exported? Maybe it could work internally only?
Also - is there any different between our current private function `getUuidFromHref`? Can they be wrapped or reused?

Reply:
This functions used widely in terraform also due that we need get ID from href -> specific vapp network handling. Private function is slight different - regular expression is slight changed.

Reply:
The only difference between the regular expressions in public and private functions is that the public one allows text after the UUID.
I think that, instead of adding a function that is 99% the same as another, we should add a parameter to match  either open ended or closed strings.
Alternatively, the two functions should have a different name, because `getUuidFromHref` matches an entity UUID (ID at the end), while `GetUuidFromHref` matches an UUID from an action HREF (ID/action at the end). 

Given that the action is discarded, I suggest this interface:

```go
func GetUuidFromHref(href string, idAtEnd bool) (string, error) {
   searchExpression := `^https://.+/([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})`
   if idAtEnd {
      searchExpression +=`$`
   } else {
      searchExpression += `.*$`
   }
   reGetID := regexp.MustCompile(searchExpression)
   // [...]
} 
```
This way, we use just one function for two needs.

Calls to `getUuidFromHref` will become `GetUuidFromHref(href, true)`, while calls to `GetUuidFromHref` will become `GetUuidFromHref(href, false)`

Reply:
I'm in favor of a common function as well to avoid duplication.

Reply:
good idea, change added.